library/src/main/java/ch/decent/sdk/model/TypeAdapters.kt

```diff
@@ -118,15 +121,25 @@ object OperationTypeFactory : TypeAdapterFactory {
         override fun read(reader: JsonReader): T? {
           val el = Streams.parse(reader)
           val idx = el.asJsonArray[0].asInt
-          val op = OperationType.values().getOrElse(idx) { OperationType.UNKNOWN_OPERATION }
-          if (op == OperationType.UNKNOWN_OPERATION) return UnknownOperation(idx) as T?
-
+          val opType = OperationType.values().getOrElse(idx) { OperationType.UNKNOWN_OPERATION }
           val obj = el.asJsonArray[1].asJsonObject
-          return op.clazz?.let {
+
+          val op = if (opType == OperationType.UNKNOWN_OPERATION) UnknownOperation(idx)
+          else opType.clazz?.let {
             val delegate = gson.getDelegateAdapter(this@OperationTypeFactory, TypeToken.get(it))
-            (delegate.fromJsonTree(obj) as BaseOperation).apply { this.type = op } as T?
-          } ?: EmptyOperation(op) as T?
+            (delegate.fromJsonTree(obj) as BaseOperation).apply { this.type = opType }
+          } ?: EmptyOperation(opType)
+
+          return op.parseCustomOp(gson) as T?
         }
+
+        private fun BaseOperation.parseCustomOp(gson: Gson): BaseOperation =
```

Comment:
Shouldn't this be responsibility of SendMessageOperation?

Reply:
i don't think so, in case we have other custom ops, i would rather keep it here in one place